### ISSUE TITLE
client got white blank screen running on windows 

### ISSUE THREAD

**Issue by Ray8716397** [rustdesk/rustdesk#418]
<img width="866" alt="屏幕截图 2022-02-27 115246" src="https://user-images.githubusercontent.com/45814273/155867741-b18da352-67c5-4adb-a178-3ecbc405ed62.png">

please help

**Comment by rustdesk** [comment]
Try out https://github.com/rustdesk/rustdesk/releases/download/tmp/rustdesk-1.1.9-putes.exe please!

**Comment by Ray8716397** [comment]
<img width="623" alt="image" src="https://user-images.githubusercontent.com/45814273/155867873-2478c26f-e481-4b6c-b261-0fd95e5dbcc9.png">


> Try out https://github.com/rustdesk/rustdesk/releases/download/tmp/rustdesk-1.1.9-putes.exe please!

still not working

**Comment by rustdesk** [comment]
How about this one https://github.com/rustdesk/rustdesk/releases/download/1.1.8/rustdesk-1.1.8-skia.zip?

**Comment by Ray8716397** [comment]
<img width="838" alt="image" src="https://user-images.githubusercontent.com/45814273/155867916-8a77560e-b114-49b3-9f4a-2303c722168c.png">
tried it before

**Comment by Ray8716397** [comment]
<img width="998" alt="image" src="https://user-images.githubusercontent.com/45814273/155867986-6062d7ab-fa61-492c-960f-fb15badf00ce.png">
seems like no error log at all.

**Comment by rustdesk** [comment]
> <img alt="image" width="998" src="https://user-images.githubusercontent.com/45814273/155867986-6062d7ab-fa61-492c-960f-fb15badf00ce.png">
> 
> seems like no error log at all.

Embarassing.

**Comment by rustdesk** [comment]
Can you build and run from source code?

**Comment by Ray8716397** [comment]
i‘ll try,no rust exprience though.

**Comment by rustdesk** [comment]
> i‘ll try,no rust exprience though.

Thanks, if any problem please ask.

**Comment by rustdesk** [comment]
@Ray8716397 can you build and run this one? It only depends on rust-sciter. Please install rust first https://rustup.rs/. Then run it in git bash, `cargo run`

https://gitlab.com/rustdesk1/rustdesk-debug.git

**Comment by Ray8716397** [comment]
<img width="1190" alt="image" src="https://user-images.githubusercontent.com/45814273/155917379-8e8e5ac7-a65d-4fc4-8997-0b3cded2e36b.png">


> @Ray8716397 can you build and run this one, https://gitlab.com/rustdesk-official/rustdesk-debug.git? It only depends on rust-sciter. Please install rust first https://rustup.rs/. Then run it in git bash, `cargo run`.

**Comment by rustdesk** [comment]
Thank you so much, can you replace src/index.html with a very simple one, and run again? like this
`
<div>xxx</div>
`

**Comment by Ray8716397** [comment]
？how ，sorry  i dont undersand

> Thank you so much, can you replace src/index.html with a very simple one, and run again? like this `xxx`

**Comment by rustdesk** [comment]
> ？how ，sorry i dont undersand

Change the content of https://gitlab.com/rustdesk-official/rustdesk-debug/-/blob/main/src/index.html to `xxx`

**Comment by Ray8716397** [comment]
<img width="1025" alt="image" src="https://user-images.githubusercontent.com/45814273/155919175-13d9ff37-9d5b-43f9-b363-54b3826fadd0.png">

**Comment by rustdesk** [comment]
Could you please `git pull`? you will see a new directory `test`. There is an executable `scapp.exe`. Can you run it and test `index.html` and `xxx.html` under the same directory?

**Comment by Ray8716397** [comment]
<img width="802" alt="image" src="https://user-images.githubusercontent.com/45814273/155923906-a47a940a-4355-4c48-8eae-bff114f2a11a.png">

**Comment by rustdesk** [comment]
@c-smile any input?

**Comment by rustdesk** [comment]
> <img alt="image" width="802" src="https://user-images.githubusercontent.com/45814273/155923906-a47a940a-4355-4c48-8eae-bff114f2a11a.png">

Could you tell us the system info of your Windows? Hope @c-smile the autor of Sciter can give some advice.

**Comment by Ray8716397** [comment]
<img width="561" alt="image" src="https://user-images.githubusercontent.com/45814273/155925234-00bebf97-07cd-46ef-8d25-cef5d76d5f72.png">

**Comment by rustdesk** [comment]
> <img alt="image" width="561" src="https://user-images.githubusercontent.com/45814273/155925234-00bebf97-07cd-46ef-8d25-cef5d76d5f72.png">

Thanks a lot

**Comment by Ray8716397** [comment]
is there a command line version i can run on windows linux sub system?

**Comment by rustdesk** [comment]
> is there a command line version i can run on windows linux sub system?

Yes, https://github.com/rustdesk/rustdesk/blob/5af12584546518be28adce04b50796a080d4d576/src/main.rs#L110 
But I have not maintained it any more.

**Comment by c-smile** [comment]
Please try the same but from folder/path that contains only ASCII symbols. What you will get?
I suspect problem is in incorrect file:// url encoding so asking.

**Comment by rustdesk** [comment]
> ASCII

Not make sense, at least Scapp works fine with non-ASCII path on my Windows.

**Comment by c-smile** [comment]
May I get exact path where scapp fails (to test it here).

**Comment by rustdesk** [comment]
> test it here

I can not reproduce this problem on my Windows. Only @[Ray8716397](https://github.com/Ray8716397) can

**Comment by dinojs** [comment]
Facing the same issue on Windows 11.

**Comment by rustdesk** [comment]
> Facing the same issue on Windows 11.

Could you run from source and get some clue? 
https://rustdesk.com/docs/en/dev/build/windows/

**Comment by bay-kah** [comment]
This may help, 

I get this issue. But also with teamviewer and edge on the same machine. When moving the window edge briefly shows it content then disappears. I am looking to replace teamviewer with rust desk. I'm going to guess its some kind of hardware acceleration issue.

**Comment by cmucgit** [comment]
Any news on this? I'm having this exact same issue in 2 of my machines

**Comment by rustdesk** [comment]
Rewriting with Flutter desktop.

**Comment by cmucgit** [comment]
Ok, thanks. Just a quick note for others, I saw the solution somewhere else, downloading and installing this driver fixed the issue for now:
https://www.amyuni.com/downloads/usbmmidd_v2.zip

**Comment by rustdesk** [comment]
> https://www.amyuni.com/downloads/usbmmidd_v2.zip

Interesting， why is it related to a driver?

**Comment by cmucgit** [comment]
In my cases, I'm trying to run RustDesk on virtual machines and all in one company machines that I use kind of as remote console or KVM of some sort to access my servers. These machines have no monitors plugged in and I get the blank white screen in RustDesk. But when I install the above monitor driver, it works

**Comment by c-smile** [comment]
By default Sciter works in H/W accelerated mode - that requires video drivers to be present.
It also can operate without video drivers - by using software rasterization. 
But that's need to be set on application side.

**Comment by rustdesk** [comment]
> software rasterization

We have set this per you suggestion https://github.com/rustdesk/rustdesk/issues/132#issuecomment-886069737.

https://github.com/rustdesk/rustdesk/blob/92f1f17ca2b95fe11b9e1788333cab6de821884b/src/ui.rs#L82

**Comment by wneiner** [comment]
This issue happened for me during testing to see if this was good to replace teamviewer.  I couldn't get it to show anything in the window.  In troubleshooting I found that it was Chrome Remote Desktop session that wasn't allowing it to be visible.  I closed that and went in through an RDP session and the window was still sitting there but this time it is showing the information, no need to relaunch. Only difference of chrome remote and rdp.

Now it just needs UAC support, and this will be golden.

Thank you,

**Comment by grummbeer** [comment]
@Ray8716397 @dinojs have you tried the nightly version?  https://github.com/rustdesk/rustdesk/releases/tag/nightly

Do you still have the white screen? 

probably same as #80

**Comment by SystemFuchs** [comment]
A by far better solution for modern Windows versions is the installation of a Indirect Display Driver (IDD). You can find a working sample here (only sharing the link, not my software). [https://github.com/ge9/IddSampleDriver](https://github.com/ge9/IddSampleDriver). 

If you are running a headless Windows system on kvm just install the QXL virtual GFX card & you should be fine.

**Comment by rustdesk** [comment]
> a Indirect Display Driver (IDD)

We have done it, but we do not EV certificate to run it.

**Comment by SystemFuchs** [comment]
The guy behind the above link has deployed the driver with the windows package manager [Scoop](https://scoop.sh/)

```
scoop bucket add extras
scoop bucket add nonportable
scoop install iddsampledriver-ge9-np -g
```

I've tested this & when you install it this way the driver seems to be signed (maybe self signed). But it does work,
here is also the thread in the original repo where the cert deployment is discussed: 

[https://github.com/roshkins/IddSampleDriver/issues/26](https://github.com/roshkins/IddSampleDriver/issues/26)

**Comment by rustdesk** [comment]
But we need more control, e.g. change  resolution, disable/enable it when monitor remove/add etc.

**Comment by rustdesk** [comment]
> The guy behind the above link has deployed the driver with the windows package manager [Scoop](https://scoop.sh/)
> 
> ```
> scoop bucket add extras
> scoop bucket add nonportable
> scoop install iddsampledriver-ge9-np -g
> ```
> 
> I've tested this & when you install it this way the driver seems to be signed (maybe self signed). But it does work, here is also the thread in the original repo where the cert deployment is discussed:
> 
> [roshkins/IddSampleDriver#26](https://github.com/roshkins/IddSampleDriver/issues/26)

@fufesou check this, may be we do not have to EV to use our ldd driver. I do not think a sample driver has EV signed.

**Comment by SystemFuchs** [comment]
If you compile the drive with VS it has a options txt. In this file all resolutions are defined & can be changed through the windows system tool. 
![image](https://user-images.githubusercontent.com/3988026/221553346-945b6cd8-35bc-494e-ba12-878fc369f4f7.png)
![image](https://user-images.githubusercontent.com/3988026/221553411-005ad2b8-5748-4e08-8d2d-8e6046672511.png)
![image](https://user-images.githubusercontent.com/3988026/221553579-0cdc905c-7ee8-44ae-9184-1638dfcfb148.png)

**Comment by rustdesk** [comment]
@SystemFuchs Thanks

**Comment by fufesou** [comment]
@SystemFuchs Thanks. I've tried install the driver manually. But I find the released package may be already signed.
I'm not familiar with signing. But we may need more steps to make our driver signed.

And I've asked the author what to do next to sign the driver. https://github.com/roshkins/IddSampleDriver/discussions/33

https://github.com/roshkins/IddSampleDriver/issues/26
![7c71ed9cd26fb977d621a78502bf507](https://user-images.githubusercontent.com/13586388/222942431-b84f450e-058d-4176-8c3c-73182582a4a8.png)

https://github.com/fufesou/RustDeskIddDriver/releases
![80ff36ef044de07def5abe15549b10c](https://user-images.githubusercontent.com/13586388/222942432-f82a458c-e4b2-4161-a991-e5f8873b2966.png)

**Comment by fufesou** [comment]
I've found the way to create cert file. I'll test later.
https://learn.microsoft.com/en-us/windows-hardware/drivers/install/creating-test-certificates

**Comment by fufesou** [comment]
@rustdesk I can sign our driver with a test cert. But I'm not sure if it meets our current needs.
Maybe we can provide a prompt window for user to trust and install the cert?

**Comment by rustdesk** [comment]
go ahead with test cert first

**Comment by Retapich** [comment]
I also encountered this today when I upgraded from **Win8.1 Embedded Industry** to **Win10 LTSC 2019** on an old laptop.

The problem was in the latest version of the driver for the GPU **Nvidia GeForce 710M**
(_it is possible that this can also be manifested on other GPU models built on the same [Fermi 2.0](https://www.techpowerup.com/gpu-specs/geforce-710m.c2017#:~:text=N14M%2DGL-,Architecture,-Fermi%202.0) architecture_).
The latest available driver version for this GPU "**391.35**" causes this problem.

The solution to this problem was to install the previous version of the driver "**391.24**".
But I think I have found the root cause of this problem.
At least on the previous version of the driver, it causes this problem again under the following circumstances⤵️

Go to the "**Nvidia Control Panel**" in the options called "**Manage 3D Settings**", and switch the "**Preferred graphics processor:**" option from "**Auto-select**" to "**Nvidia performance...**" (_this is a rough name for this the option, because I don't remember exactly. I took the screenshot below a couple of hours before writing this comment_)⤵️
![2023-06-17_164126](https://github.com/rustdesk/rustdesk/assets/19671942/4fbb189b-10a7-484f-8b8a-585b05433d25)
### After applying these settings, the problem appears.

Currently, I have installed an even older version of the GPU driver "**388.71**", because the video game "**Roblox**" (which my nephew regularly plays on this laptop), like **RustDesk**, refuses to work, showing a solid white window even on the driver version "**391.24**" on which **RustDesk** works flawlessly.

**Comment by japri** [comment]
I found the fix,

download this driver --> [(https://www.amyuni.com/downloads/usbmmidd_v2.zip)]

unpack run install with this command administrator elevated--> deviceinstaller64 install usbmmidd.inf usbmmidd

and then this command --> deviceinstaller64 enableidd 1

done.

There is complete manual in the zip. please read it.
